conans/client/build/autotools_environment.py

```diff
@@ -93,13 +89,22 @@ def _get_host_build_target_flags(self):
         else:
             target = None
 
-        if os_detected is None or arch_detected is None or self._arch is None or self._os is None:
+        settings_build = getattr(self._conanfile, 'settings_build', None)
+        if settings_build:
+            os_build, arch_build = get_build_os_arch(self._conanfile)
```

Comment:
Maybe don't call again the method that knows how to fallback, because you already checked that you have settings_build.

Reply:
I need to check here if a have `settings_build` because the values returned by the fallback are not the ones expected here. 

Probably this is a bug in the existing code: this function doesn't consider `settings.os_build` or `settings.arch_build`, only those detected of platform.xxx().